kotlin/src/test/kotlin/io/smallrye/mutiny/coroutines/DeferredAsUniTest.kt

```diff
@@ -78,4 +84,106 @@ class DeferredAsUniTest {
             assertThat(subscriber.awaitItem().item).isNull()
         }
     }
+
+    @Test
+    fun `test uni cancellation before deferred completes`() {
+        runBlocking {
+            // Given
+            val deferredLatch = CountDownLatch(1)
+            val deferred = async {
+                delay(10_000) // simulate long running job
+                "Hello, World!"
+            }
+            deferred.invokeOnCompletion {
+                deferredLatch.countDown()
+            }
+            val uniLatch = CountDownLatch(1)
+            val uni = deferred.asUni()
+                .onCancellation().invoke { uniLatch.countDown() }
+
+            // When
+            val subscriber = UniAssertSubscriber.create<String>()
+            uni.subscribe().withSubscriber(subscriber)
+            subscriber.cancel()
+
+            // Then
+            awaitLatch("Uni", uniLatch)
+            awaitLatch("Deferred", deferredLatch)
+            assertThat(deferred.isCancelled)
+                .`as`("Check that Deferred is cancelled")
+                .isTrue()
+        }
+    }
+
+    @Test
+    fun `test cancelling deferred does not impact siblings`() {
+        runBlocking {
+            // Start before async.
+            val child1Latch = CountDownLatch(1)
+            val child1 = launch {
+                delay(1000)
```

Comment:
as the cancellation state is check when suspend functions are called, maybe you can to a
```
(1..100).forEach {
    delay(10)
}
```
instead the long wait.

Reply:
Sure, I can change that. We'd may need to include an `isActive` check on each iteration as well.

That said, the [`delay` function](https://github.com/Kotlin/kotlinx.coroutines/blob/c112be4a896a5b937883f0a6cd2ac9a2e71e15cc/kotlinx-coroutines-core/common/src/Delay.kt#L104-L106) _should_ return immediately on cancellation:

> If the Job of the current coroutine is cancelled or completed while this suspending function is waiting, this function immediately resumes with CancellationException. There is a prompt cancellation guarantee.
> https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/delay.html

Reply:
ok, then lets believe the Kotlin documentation.
regarding the build-failure: not checked in detail, but maybe a rebase from main branch helps ;)